tests/suites/0_stateless/03_dml/03_0004_select_order_by_db_table_col_v2.sql

```diff
@@ -0,0 +1,26 @@
+SET enable_planner_v2=1;
+DROP DATABASE IF EXISTS a;
+DROP DATABASE IF EXISTS b;
+CREATE DATABASE a;
+CREATE TABLE a.t(id INT, id2 INT);
+INSERT INTO a.t VALUES (1, 1),(2, 2);
+
+SELECT id FROM a.t ORDER BY id2 ASC;
+SELECT id FROM a.t ORDER BY t.id2 ASC;
+SELECT id FROM a.t ORDER BY a.t.id2 ASC;
+
+SELECT DISTINCT(id) FROM a.t ORDER BY a.t.id;
+SELECT DISTINCT(id) FROM a.t ORDER BY a.t.id2; -- {ErrorCode 1065}
+-- this will err but if has a alias will not err.
+-- SELECT SUM(id) FROM a.t ORDER BY a.t.id2;
+-- ERROR 1105 (HY000): Code: 1006, displayText = Unable to get field named ""id2"_1". Valid fields: ["\"SUM(id)\"_4"].
+SELECT SUM(id) as id2 FROM a.t ORDER BY a.t.id2;
+-- expect err
+SELECT DISTINCT(id) as id2 FROM a.t ORDER BY a.t.id2;
```

Comment:
Here, we need an error, but now it will succeed. I will create a new ISSUE.

```
'root'@mysqldb 17:57:14 [(none)]> SELECT DISTINCT(id) FROM a.t ORDER BY a.t.id2;
ERROR 1105 (HY000): Code: 1065, displayText = error: 
  --> SQL:1:39
  |
1 | SELECT DISTINCT(id) FROM a.t ORDER BY a.t.id2
  |                                       ^^^^^^^ for SELECT DISTINCT, ORDER BY expressions must appear in select list

```